src/LanguageSwitcher/LanguageSwitcher.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import { Link } from 'gatsby'
+import LocaleContext from '../context/LocaleProvider'
+import PropTypes from 'prop-types'
+
+const LocaleComponent = ({ location }) => {
+  const lang = React.useContext(LocaleContext)
+  const i18n = lang.i18n[lang.locale]
+  return (
+    <>
+      {i18n.locale === 'en-ca' ? (
+        <Link
+          to={`/fr${
+            location.pathname.includes('/en')
+              ? location.pathname.replace('/en', '')
+              : location.pathname
+          }`}
+          className="Header__LanguageSwitcher-button"
+        >
+          FR
+        </Link>
+      ) : i18n.locale === 'fr-ca' ? (
+        <Link to="/" className="Header__LanguageSwitcher-button">
+          EN
+        </Link>
+      ) : (
+        ''
+      )}
+    </>
+  )
+}
+
+LocaleComponent.propTypes = {
+  location: PropTypes.object,
+}
+export default LocaleComponent
```

Comment:
Try adding another page like about and fr.about. You'll see that when you use language switcher on those It won't work. 

- [ ] Try with about page

Reply:
It seems to work: Just I am not sure if the url path is supposed to look like that ? /fr/about ? Example below: 

![Screen Shot 2020-08-27 at 9 42 43 AM](https://user-images.githubusercontent.com/46905898/91450160-fcd07d00-e849-11ea-8af2-0897a9b6a10d.png)